### ISSUE TITLE
Can't really serve static files with channels 

### ISSUE THREAD

**Issue by flo-dhalluin** [andrewgodwin/channels#76]
I try to transition an existing django app to channel, in dev/debug mode. 

Apparently serving static files with the default DatabaseChannelLayer quickly hogs the database, my sqlite file makes a few 10s of Megs only after a few request ( I am serving my static files through django ..  default debug/dev mode .. ). 
My understanding is that all my static assets ( large js/css app) is copied at each request in the database. Fine . But then daphne fails with : ( purging the database solve this for the next 2 requests ..). 

    daphne/http_protocol.py", line 136, in serverResponse
        raise ValueError("Got multiple Response messages!")

I can see why channels should not be used for static files and a proper production setup will use a proper webserver for static assets, but is there a hidden usable "quick and dirty for dev mode" way of serving static files with channels that I could not see in the help ? 
I guess we could have runserver spawning a separate static file server, but at least I think it should be mentionned in the doc somewhere like "DON'T serve statics with channels / DatabaseChannelLayer" 

Let me know how I could help.

**Comment by andrewgodwin** [comment]
The docs already say not to use DatabaseChannelLayer, but I'll reinforce it. It's really _not_ fast with the built in Django stuff (which chunks things into 4KB chunks to serve them!)

I'm going to make it re-chunk them up to a bigger size, but yes, try the Redis backend or the in memory backend if you're just using runserver.

**Comment by flo-dhalluin** [comment]
Ok, I noticed it's not fast, and I am good with it (after all, debug mode ..), but it is not usable at all as it crashes the server. I am talking about an angular app so quite large assets (default main angular.js file is about 1Mo, add a few more dependencies ... ). 

 I'll see if can work on daphne to handle static files directly from twisted, maybe with static_dir/url option ?

**Comment by andrewgodwin** [comment]
No, just use the in memory backend, it should be faster than normal runserver (channel layer `asgiref.inmemory.InMemoryLayer`)

**Comment by andrewgodwin** [comment]
Obviously, in production, you wouldn't serve static files through Daphne anyway - I just want to make sure `runserver` is usable.

**Comment by flo-dhalluin** [comment]
All right ! much better !  
Then why the "Quickstart" guide tell to use the Database layer ? . Obviously I a fast webserver for my statics, channels or not ...

**Comment by andrewgodwin** [comment]
It's there because it's the only one that works with multiple processes and without installing extra things (like Redis). I hope to improve the performance of static file serving on the database backend to at least acceptable levels, though - I'll either get that to an alright level or reword it to suggest redis or memory backends.

**Comment by flo-dhalluin** [comment]
Wonderful, `asgiref.inmemory.ChannelLayer` works sufficiently well for my dev/test purposes.

I don't know if it is "large" assets, parallel fetch of them or the combination, but database backed layer fails, quasi consistently.

**Comment by andrewgodwin** [comment]
The database layer needs proper transaction support to correctly do message delivery - given that you were using SQLite, it's likely that it wasn't correctly being fully serialisable and so not working!

I might just dump the database one from the docs entirely, and switch the examples to use InMemory at the start.
